### PR TITLE
 Fix taint analyzer for worker interactions and lambda

### PR DESCRIPTION
**Pull request by ayomawdb**
## Purpose
With new changes to the compiler, Ballerina workers are modeled as lambda functions. With this change, taint checking for worker interactions did not work as expected. Results of all the worker interactions were considered `untainted`. 

## Approach
With the current PR:
- Taint analyzer will analyze Lambda functions and workers as part of the analysis of the enclosing function. (This is done because Lambda functions has direct access to enclosing function's variables).
- Ballerina taint analyzer is checking the trained status of worker-send expression to determine the tainted status of worker-receive expression. 

This PR also resolves: #13368 

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java [hunk 1]
```diff
@@ -260,20 +250,25 @@ public static TaintAnalyzer getInstance(CompilerContext context) {
 
     public TaintAnalyzer(CompilerContext context) {
```
Comment (anupama-pathirage) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java]: This method seems to be a private method

Comment (ayomawdb) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java]: This is commonly `public` across all other compiler phases. Did not differ from the norm. Example: https://github.com/ballerina-platform/ballerina-lang/blob/450eb61585cce0765c2dd834fab54272653f52a1/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java#L233

Comment (ayomawdb) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java]: @sameerajayasoma @hasithaa is this something we can fix across all phases, or is there a reason for keeping the constructor `public`?

**Comment by hasithaa** [comment]
@ayomawdb , Shall we add closure variable support as we discussed offline. ? PR https://github.com/ballerina-platform/ballerina-lang/pull/14614 is merged to master.

**Comment by ayomawdb** [comment]
> @ayomawdb , Shall we add closure variable support as we discussed offline. ? PR #14614 is merged to master.

@hasithaa, made the suggested change with the latest commit.

**Pull request merged**